### DISCUSSION TITLE
[Section 3.14 Interpolação, locals e outputs] Como o terraform interpreta as variáveis em outros arquivos?

### DISCUSSION THREAD

**Discussion by LuizHenriqueEgito** [chgasparoto/curso-aws-com-terraform#158] [Q&A]
Professor, você criou um arquivo random.tf e colocou o codigo chamado resource "random_pet" "bucket" {...} nele. Depois você vai para o arquivo s3.tf e chama random_pet mas não especifica que ele vem do arquivo random.tf. Ai está minha duvida o terraform vê todos os arquivos como se estivessem em um mesmo arquivo para não precisar especificar de onde o random_pet veio? Isso ficou confuso para mim  e como funciona essa logica?

**Comment by chgasparoto** [comment]
@LuizHenriqueEgito sim, o Terraform interpreta todos os arquivos (no mesmo nível) dentro de uma pasta como se você um arquivo só. 
Isto é conhecido como "root module" https://developer.hashicorp.com/terraform/language/modules#the-root-module
